custom.js

```diff
@@ -15,7 +15,7 @@ function init (type, message, cause) {
 function CustomError (message, cause) {
   Error.call(this)
   if (Error.captureStackTrace)
-    Error.captureStackTrace(this, arguments.callee)
+    Error.captureStackTrace(this, CustomError)
```

Comment:
Does this work for subclasses of `CustomError`? Or does that fall outside intended use of `errno`?

Reply:
The following code (`foo.js`):

```js
const create = require('errno').create                      
const CustomError = require('errno').custom.CustomError     
const inherits = require('inherits')                    
                                                        
// inherits from CustomError (which inherits from Error)
const MyError = create('MyError')                       
                                                        
// same but using traditional inheritance               
function MyError2 (message, cause) {                    
  CustomError.call(this, message, cause)                
}                                                       
                                                        
inherits(MyError2, CustomError)                         
                                                        
function bar () {                                       
  console.log(new MyError())                            
  console.log(new MyError2('MyError2', 'woot'))         
}                                                       
                                                        
function foo () {                                       
  bar()                                                 
}                                                       
                                                        
foo()                                                   
```

Yields the following output:

```
MyError
    at bar (/home/lms/src/node-errno/foo.js:16:15)
    at foo (/home/lms/src/node-errno/foo.js:21:3)
    at Object.<anonymous> (/home/lms/src/node-errno/foo.js:24:1)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Function.Module.runMain (module.js:682:10)
    at startup (bootstrap_node.js:191:16)
CustomError: MyError2
    at new MyError2 (/home/lms/src/node-errno/foo.js:10:15)
    at bar (/home/lms/src/node-errno/foo.js:17:15)
    at foo (/home/lms/src/node-errno/foo.js:21:3)
    at Object.<anonymous> (/home/lms/src/node-errno/foo.js:24:1)
    at Module._compile (module.js:641:30)
    at Object.Module._extensions..js (module.js:652:10)
    at Module.load (module.js:560:32)
    at tryModuleLoad (module.js:503:12)
    at Function.Module._load (module.js:495:3)
    at Function.Module.runMain (module.js:682:10)
```

And the output is identical to current master and this PR. So it the inherited constructor is part of the callstack in this case.

Reply:
I guess `MyError2` would have to do `Error.captureStackTrace(this, MyError2)` in this case, to get rid of the last stack frame. Is there any way around this? It would be nice if inherited objects could get this automatically.

Reply:
Maybe `Error.captureStackTrace(this, this.constructor)`?

Reply:
It works :)

Reply:
Updated.

Reply:
I should write some tests for this, e.g. use the code above, check `.stack` and parse out the functions.

Reply:
Tests added.